Dynavity/Dynavity/view/canvas/CanvasView.swift

```diff
@@ -1,21 +1,97 @@
 import SwiftUI
 
 struct CanvasView: View {
-    @ObservedObject var canvas: Canvas
+    @ObservedObject var viewModel = CanvasViewModel()
+
+    // For drag interactions
+    @State var portalPosition = CGPoint()
```

Comment:
Can this be renamed to `mapViewPositionFromCentre` or something else if it's too long? I think it can be quite hard to understand what `portal` refers to